src/NuGet.Core/NuGet.LibraryModel/GlobalSuppressions.cs

```diff
@@ -25,3 +25,4 @@
 [assembly: SuppressMessage("Build", "CA2227:Change 'Items' to be read-only by removing the property setter.", Justification = "<Pending>", Scope = "member", Target = "~P:NuGet.LibraryModel.Library.Items")]
 [assembly: SuppressMessage("Build", "CA2227:Change 'NoWarn' to be read-only by removing the property setter.", Justification = "<Pending>", Scope = "member", Target = "~P:NuGet.LibraryModel.LibraryDependency.NoWarn")]
 [assembly: SuppressMessage("Build", "CA1067:Type NuGet.LibraryModel.FrameworkDependency should override Equals because it implements IEquatable<T>", Justification = "<Pending>", Scope = "type", Target = "~T:NuGet.LibraryModel.FrameworkDependency")]
+[assembly: SuppressMessage("Globalization", "CA1308:Normalize strings to uppercase", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.LibraryModel.LibraryIncludeFlagUtils.GetFlags(System.Collections.Generic.IEnumerable{System.String},System.Boolean)~NuGet.LibraryModel.LibraryIncludeFlags")]
```

Comment:
Putting on my API design review hat on, do we need this new API?

Instead, can the caller add `buildTransitive` to the `IEnumerable<string>` before calling the existing public API?

Not that I'm an expert on `NuGet.LibraryModel`, but my gut reaction is that it's a bit of a leaky abstraction to add the bool parameter, and therefore might not be a good API design.

Reply:
The problem is that the assets file contains only `BuildTransitive` and when its read by pack, the value comes back as both.  In this code path, we are going to be using the value in the `.nuspec` so it should really only contain what the user specified.  Since its a transitive dependency, `Build` should not flow unless specified right?  

Given that, I couldn't come up with a better API change, I need this method to return the true values.  I am open to suggestions though so please let me know.

Reply:
> The problem is that the assets file contains only BuildTransitive and when its read by pack, the value comes back as both

That sounds like a bug to me.

When I do `<PackageReference Include="Package1" Version="1.0.0" IncludeAssets="compile;runtime;build;buildtransitive" />` and restore, in the assets file I see `"include": "Runtime, Compile, Build, BuildTransitive"`.  So, the assets file does seem capable of specifying `Build` and `BuildTransitive` independently.

When I try packing a project (that does not use CPM) with PackageReferences modifiers like `PrivateAssets="none"`, `PrivateAssets="analyzers"`, or the above IncludeAssets, the generated nuspec contains the include and exclude attributes that I'd expect, including being capable of specifying `Build` and `BuildTransitive` independently.

This has me wondering, is this PR trying to solve the problem for CPM in a different way that it's already implemented for non-CPM projects?

Perhaps I don't understand your comment very well. I certainly don't know the code being modified, so if there are indeed practical constraints that aren't obvious from a theoretical knowledge of the asset selection feature, then I'm unaware. But I don't understand why reading the assets file will parse `BuildTransitive` as both `Build` and `BuildTransitive`, when the assets file is capable of representing both. From a pack point of view, it also doesn't appear to be a limitation of non-CPM projects.

Reply:
There's a related bug: https://github.com/NuGet/Home/issues/9672

Build transitive is implemented in a way that might not be the most intuitive. 
build and buidltransitive aren't exclusive like all the other asset types. 

if a package has build and buildtransitive files, only buildtransitive will be used. 

By extension, if you write `PrivateAssets="buildtransitive"`, pretty sure it tries to suppress both build and buildtransitive

Reply:
Yeah, this is super confusing. Accordign to my tests this reference:
```xml
<PackageReference Include="PackageWithBuildTransitive" Version="1.0.0" IncludeAssets="Build" ExcludeAssets="BuildTransitive" />
```
doesn't include `Build` assets although it is explicitly requested.

Reply:
Probably getting off-topic for this PR now, but does this also mean that the use case for https://github.com/NuGet/Home/issues/6938 isn't going to work?  Even with that feature implemented, the package trying to include its dependency's `build` assets isn't going to work as expected?

Reply:
So what's the consensus here?  With my code change, I feel that i get the expected result.  Without changing it, it stays as confusing as ever.  It only applies when someone tries to exclude `BuildTransitive`  independently which I would imagine isn't too common.

Reply:
I think it'd be great if the build/buildtransitive chnage is fixed in all scenarios. 
Having it change only for transitive pinning is a probably not enough?